File: week3/8bitbully/hasCycle.cpp
// 141. Linked List Cycle

struct ListNode {
  int val;
  ListNode *next;
  ListNode(int x) : val(x), next(nullptr) {}
};

class Solution {
public:
  bool hasCycle(ListNode *head) {

    int MAX_ITER = 1e4;

    while (head && MAX_ITER--) {
      head = head->next;
    }

    return head;
  }
};

File: week3/MarshalLee/HasCycle.cs
using System;

public bool HasCycle(ListNode head)
{
    HashSet<ListNode> visited = new();
    ListNode currentNode = head;
    while (currentNode != null)
    {
        if (visited.Contains(currentNode))
        {
            return true;
        }
        visited.Add(currentNode);
        currentNode = currentNode.next;
    }
    return false;
    //if (head == null) return false;

    //ListNode fast = head;
    //ListNode slow = head;

    //while (fast.next != null && fast.next.next != null)
    //{
    //    fast = fast.next.next;
    //    slow = slow.next;

    //    if (fast == slow) return true;
    //}
    //return false;
}
